cmd/podman/machine/machine.go

```diff
@@ -64,3 +96,63 @@ func getMachines(toComplete string) ([]string, cobra.ShellCompDirective) {
 	}
 	return suggestions, cobra.ShellCompDirectiveNoFileComp
 }
+
+func initMachineEvents(cmd *cobra.Command, _ []string) error {
```

Comment:
Should we really open all sockets at init time and keep them open until we end?
For example `podman machine ssh` can be a very long running command and does not need the socket all AFAICT.

I think it would be better to only open the sockets when we actually want to write to them.

Reply:
@Luap99 I was holding them open as commands like init can fire off multiple events and it seemed more expensive to keep opening/closing vs. holding them open. 

I am hoping to here if @benoitf is interested in events for ssh etc.

Reply:
How about moving this into a sync.Once in newMachineEvent. In this case we would open the sockets only once and only when we need it.

Reply:
This was not addressed, you still open all sockets at init time, the added sync once is a nop